### PR TITLE
Add approvals for specific risks screen

### PR DESCRIPTION
**Pull request by bryangaledxw**
# Changes in this PR

This PR adds a new screen for adding approvals information, on the basis of displayed risk information

As part of this, we update our risk widgets from https://github.com/ministryofjustice/hmpps-risk-ui-elements to better handle reported API errors

### Task list entry
localhost_3000_applications_aec6acd2-9931-472d-ada4-8d4ef8d0db10

### Page

localhost_3000_applications_339ad3eb-ca05-4772-81e8-649a8c695e44_tasks_approvals-for-specific-risks_pages_approvals-for-specific-risks

### Errors
localhost_3000_applications_aec6acd2-9931-472d-ada4-8d4ef8d0db10_tasks_approvals-for-specific-risks_pages_approvals-for-specific-risks
localhost_3000_applications_aec6acd2-9931-472d-ada4-8d4ef8d0db10_tasks_approvals-for-specific-risks_pages_approvals-for-specific-risks (1)

### Check your answers
localhost_3000_applications_aec6acd2-9931-472d-ada4-8d4ef8d0db10_tasks_check-your-answers_pages_review (1)

### Widgets without data
localhost_3000_applications_39491f0e-f2e0-4f9f-ae9a-f2547cbfb409_tasks_approvals-for-specific-risks_pages_approvals-for-specific-risks
localhost_3000_applications_39491f0e-f2e0-4f9f-ae9a-f2547cbfb409_tasks_check-your-answers_pages_review

# Release checklist

[Release process documentation](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/edit-v2/4247847062?draftShareId=a1c360ab-bd31-4db1-aae3-7cc002761de9).

## Pre-merge checklist

- [ ] Are any changes required to dependent services for this change to work?
  (eg. CAS API)
- [ ] Have they been released to production already?

## Post-merge checklist

- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to test
- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to preprod
- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to prod

### CODE REVIEW

**Comment by bryangaledxw** [comment]
@pezholio @richpjames "Update risk widgets and mapApiPersonRisksForUi()" is the commit you may want to take a look at, Basically I've pulled over the error handling from the widgets at https://github.com/ministryofjustice/hmpps-risk-ui-elements/

Something I feel uncertain on is how to interpret the "not_found" status code from the API. Most of the widgets don't make a distinction between "error" and "not_found". The original RoSH widget though had a "A RoSH summary has not been completed for this individual" path, and I've mapped this to "not_found". @adamhughes86 , know at all if this is accurate?

In general, I'd rather get this wrong in the direction of "showing an error when there are no risks" rather than "showing there are no risks when there's an error"

**Comment by bryangaledxw** [comment]
[Add approvals for specific risks screen](https://trello.com/c/P5lTKu96/1268-add-approvals-for-specific-risks-screen)

**Comment by bryangaledxw** [comment]
@selina-dxw Do we have text for "Check with your [who would be best] if you are unsure."?

File: server/views/components/riskWidgets/mappa-widget/template.njk [hunk 1]
```diff
@@ -1,5 +1,14 @@
...
     <p class="govuk-body-m">Multi-agency public protection arrangements</p>
-    {% if params.level %}<p class="govuk-hint govuk-body-m">Last updated: {{ params.lastUpdated | default("Not known") }}</p>{% endif %}
+    {% if params.value.level %}
+      <p class="govuk-hint govuk-body-m">Last updated: {{ formatDate(params.value.lastUpdated) if params.value.lastUpdated else "Not known" }}</p>
+    {% endif %}
+  {% else %}
+    <h3 class="govuk-heading-m"><strong>UNKNOWN</strong> MAPPA</h3>
+    <p class="govuk-body-m">Multi-agency public protection arrangements</p>
+    <hr class="govuk-section-break govuk-section-break--m govuk-section-break--visible">
+    <p class="govuk-body-m">Something went wrong. We are unable to show MAPPA information at this time. Try again later.</p>
```
Comment (tahb) [inline server/views/components/riskWidgets/mappa-widget/template.njk]: I am wondering about our suggestion of "try again" here. I think there is a higher chance that this person hasn't got the necessary data upstream and a lower chance that our request isn't honoured (retry able). If there is no data there's little value in asking the user to retry and invites the question of: "what constitutes a retry, should I start the referral again?", which is something we probably don't want to suggest.

How about we reframe this towards: 

> Something went wrong. We are unable to show or copy MAPPA information at this time. This risk data must be checked manually outside of this service.

My hope is that points the user into continuing their referral on the understanding both they and any assessors will need to go else where for the risk data. 

The same pattern of question applies to our other widgets.

**Review by tahb — APPROVED** [review]
One concern about inviting the user to try again when there's no risk data which I think warrants a conversations, otherwise it good to me.

File: server/views/components/riskWidgets/mappa-widget/template.njk (hunk 1 above)
Comment (bryangaledxw) [reply server/views/components/riskWidgets/mappa-widget/template.njk]: Good shout, and I like that wording. Retry here would be "start the application again", given that it's attached to the application at creation time

**Comment by selina-dxw** [comment]
Hi Bryan, 

There's not a clear cut role that someone could check with about approvals so I suggest removing this whole sentence: 

"Check with your [who would be best] if you are unsure"

**Pull request merged**